ui/components/src/TimeChart/TimeChart.tsx

```diff
@@ -276,10 +276,32 @@ export const TimeChart = forwardRef<ChartInstance, TimeChartProps>(function Time
     if (tooltipPinnedCoords !== null && lastTooltipPinnedCoords !== null) {
       if (!isEqual(lastTooltipPinnedCoords, tooltipPinnedCoords)) {
         setTooltipPinnedCoords(null);
-      }
-      const isCrosshairPinned = checkCrosshairPinnedStatus(seriesMapping);
-      if (tooltipPinnedCoords !== null && isCrosshairPinned) {
-        seriesMapping.pop();
+
+        const isCrosshairPinned = checkCrosshairPinnedStatus(seriesMapping);
+        if (tooltipPinnedCoords !== null && isCrosshairPinned) {
+          seriesMapping.pop();
+        }
+
+        const syncedChartPointInGrid = getPointInGrid(
```

Comment:
This branch is a quick hacky attempt to show how this feature could work (I need to clean up how pinned crosshairs are reset and add a shared util), but the key idea is using `lastTooltipPinnedCoords.plotCanvas` from ChartsProvider to update adjacent charts inside of this useEffect